### DISCUSSION TITLE
Sharing an older printer via Airprint with Cups 2.4 without generating Avahi service files

### DISCUSSION THREAD

**Discussion by BlacCello** [OpenPrinting/cups#369] [Q&A] [answered]
Hi, I am trying to share an older printer via CUPS/Airprint.
The base logic is used from [cups-avahi-airprint](https://github.com/chuckcharlie/cups-avahi-airprint) with starting avahi, cups and genering the avahi service printer files to make the printer available in the network.

I stumbled across this topic because A4 was not working anymore without modifying additional settings (PAPERSIZE env var or /etc/papersize). 
But this lead to some follow up questions for me:
[The Arch Wiki](https://wiki.archlinux.org/title/CUPS/Troubleshooting#Issues_Relating_to_Upgrade_2.3.3_-%3E_2.4.0 ) states that the **avahi service files are not required anymore**, with the introduction of Airprint support in Cups 2.4. 
However, I could not find any information on how to achieve to share the printer without generating those files. If I don't generate them, the printer won't be visible from the iPhone.

Can you give me more insights how the new feature works? 

Are there any additional configuration steps required for Cups?
Does the avahi-daemon still needs to be running, and how do I have to configure it?

**Comment by michaelrsweet** [comment, marked as answer]
@BlacCello CUPS 2.4's printer sharing automatically includes everything that iOS needs for AirPrint. Just share the printer from CUPS (and have printer sharing enabled) and it will show up. The commands would be:

```
cupsctl --share-printers
lpadmin -p NAME -o printer-is-shared=true
```

**Comment by BlacCello** [comment]
Thanks for the answer. Now I got it working after  you confirmed that this should work out of the box :+1: 
The issue was probably the missing dbus configuration.

For anyone else who is interested how this now looks like: https://github.com/BlacCello/cups-airprint
